### PR TITLE
fix for loading with missing event_id

### LINKED ISSUE #109: read_connectivity crashes with KeyError when reading older files
#### Describe the bug
When trying to load a file saved about a year ago, `read_connectivity` crashes (traceback below).

#### Steps to reproduce
Here's a file that fails this way: http://dan.mccloy.info/f2f-f2f_009-attend-theta-band-7sec-envelope-correlation.nc

```py
from mne_connectivity import read_connectivity
read_connectivity('f2f-f2f_009-attend-theta-band-7sec-envelope-correlation.nc')
```

#### Expected results
the file is loaded.

#### Actual results
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/Documents/academics/research/ilabs/ftof/connectivity/analysis/make-xarrays.py in <module>
     78                 conn_fname = (f'{slug}-{n_sec}sec-envelope-correlation.nc')
     79                 conn_fpath = os.path.join(conn_dir, conn_fname)
---> 80                 conn = mne_connectivity.read_connectivity(conn_fpath)
     81                 # envelope correlations
     82                 conn_matrix = conn.get_data('dense').squeeze()

/opt/mne/connectivity/mne_connectivity/io.py in read_connectivity(fname)
     99 
    100     # get the data as a new connectivity container
--> 101     conn = _xarray_to_conn(conn_da, cls_func)
    102     return conn

/opt/mne/connectivity/mne_connectivity/io.py in _xarray_to_conn(array, cls_func)
     45 
     46     # write event IDs
---> 47     event_id_keys = np.atleast_1d(array.attrs.pop('event_id_keys')).tolist()
     48     event_id_vals = np.atleast_1d(array.attrs.pop('event_id_vals')).tolist()
     49     event_id = {key: val for key, val in zip(event_id_keys, event_id_vals)}

KeyError: 'event_id_keys'
```

#### Additional information
looks like it was caused by #58 if I had to venture a guess.

### PR DESCRIPTION
**Pull request by drammock**
PR Description
--------------

closes #109.  There should probably be a test too, but I didn't even see a `tests/test_io.py` file and at the moment I haven't time to do a proper job of *that*.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-connectivity/blob/main/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_connectivity/io.py [hunk 1]
```diff
@@ -44,10 +44,13 @@ def _xarray_to_conn(array, cls_func):
     metadata = _prepare_read_metadata(metadata)
 
     # write event IDs
-    event_id_keys = np.atleast_1d(array.attrs.pop('event_id_keys')).tolist()
-    event_id_vals = np.atleast_1d(array.attrs.pop('event_id_vals')).tolist()
-    event_id = {key: val for key, val in zip(event_id_keys, event_id_vals)}
-    array.attrs['event_id'] = event_id
+    if 'event_id_keys' in array.attrs and 'event_id_vals' in array.attrs:
```
Comment (adam2392) [inline mne_connectivity/io.py:47]: Can you add a comment above noting that storing events are optional in connectivity?

**Review by adam2392 — APPROVED** [review]
LGTM. I don't think a test is absolutely necessary atm. 

FWIW io is tested inside the `test_connectivity.py` file, which ensures roundtrips

File: mne_connectivity/io.py (hunk 1 above)
Comment (drammock) [inline mne_connectivity/io.py:47]:
```suggestion
    # storing events is optional, not all files will have them
    if 'event_id_keys' in array.attrs and 'event_id_vals' in array.attrs:
```

**Comment by drammock** [comment]
This CI error: https://github.com/mne-tools/mne-connectivity/actions/runs/3423420815/jobs/5701993749#step:10:54 should have been fixed with matplotlib 3.6.2 (released 7 days ago). It's caused by a change in PySide 6.4, which is quite recent (maybe 1 month old?).  I'll push the same fix from https://github.com/mne-tools/mne-python/pull/11246

**Comment by drammock** [comment]
weird, the updated spec in `requirements.txt` isn't getting followed?  @larsoner any idea what's wrong here?

https://github.com/mne-tools/mne-connectivity/actions/runs/3423420815/jobs/5701993749#step:5:40

**Comment by larsoner** [comment]
For the 3.7 build I'm guessing there is no fixed wheel, maybe it's 3.8 only? In any case it installs 3.5.something.

I would just bump this build to 3.8. The failures in other builds look legitimate and related like
```
_________ test_time_resolved_spectral_conn_regression[multitaper-plv] __________
mne_connectivity/spectral/tests/test_spectral.py:590: in test_time_resolved_spectral_conn_regression
    assert_array_almost_equal(conn_data, test_conn)
E   AssertionError: 
E   Arrays are not almost equal to 6 decimals
E   
E   Mismatched elements: 35988 / 36000 (100%)
E   Max absolute difference: 0.02373527
E   Max relative difference: 0.08923911
E    x: array([[[[0.390603, 0.393248, 0.395886, ..., 0.179899, 0.[176](https://github.com/mne-tools/mne-connectivity/actions/runs/3432799079/jobs/5722434609#step:11:177)772,
E             0.173664],
E            [0.24873 , 0.250642, 0.252551, ..., 0.091652, 0.089639,...
E    y: array([[[[0.396488, 0.399161, 0.401826, ..., 0.[179](https://github.com/mne-tools/mne-connectivity/actions/runs/3432799079/jobs/5722434609#step:11:180)551, 0.176434,
E             0.173338],
E            [0.257082, 0.259085, 0.261087, ..., 0.090343, 0.088305,...
```

**Comment by drammock** [comment]
> For the 3.7 build I'm guessing there is no fixed wheel, maybe it's 3.8 only? In any case it installs 3.5.something.

To me the problem is not the matplotlib version, it's that I specify `PySide6!=6.3.0,!=6.4.0` and it installs 6.4.0 anyway.

> The failures in other builds look legitimate and related

Legitimate, yes. Related to this PR's changes? I'm not so sure.

**Comment by adam2392** [comment]
Thanks @drammock !

**Pull request merged**